airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -34,9 +35,38 @@ class TriggerRuleDep(BaseTIDep):
     IGNOREABLE = True
     IS_TASK_DEP = True
 
+    @staticmethod
+    def bake_dep_status_query():
+        TI = airflow.models.TaskInstance
+        # TODO(unknown): this query becomes quite expensive with dags that have many
+        # tasks. It should be refactored to let the task report to the dag run and get the
+        # aggregates from there.
+        q = BAKED_QUERIES(lambda session: session.query(
+            func.coalesce(func.sum(case([(TI.state == State.SUCCESS, 1)], else_=0)), 0),
```

Comment:
Can you provide me this query in SQL format? I think it can be optimized for PostgresQL by using COUNT...FILTER syntax. However, this also requires checking if this syntax has an effect on performance, or is it just syntactic sugar.
https://www.postgresql.org/docs/9.4/sql-expressions.html#SYNTAX-AGGREGATES

Reply:
Yeah, I was wondering about having some DB specific optimizations in places. Somewhere else in the Scheduler we can speed it up by doing `UPDATE task_instance ... RETURNING *` to avoid a second query, I think it helped a bit.

Reply:
May not be needed after #4751 anyway.